database/src/service/read.rs

```diff
@@ -47,145 +49,163 @@ type ResponseSend = tokio::sync::oneshot::Sender<ResponseResult>;
 /// Calling [`tower::Service::call`] with a [`DatabaseReadHandle`] & [`ReadRequest`]
 /// will return an `async`hronous channel that can be `.await`ed upon
 /// to receive the corresponding [`Response`].
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub struct DatabaseReadHandle {
-    /// Sender channel to the database read thread-pool.
+    /// Handle to the custom `rayon` DB reader thread-pool.
+    ///
+    /// Requests are "spawn"ed in this thread-pool, and responses
+    /// are returned via a channel we (the caller) provide.
+    pub(super) pool: Arc<rayon::ThreadPool>,
+
+    /// Access to the database.
+    pub(super) db: Arc<ConcreteEnv>,
+}
```

Comment:
Each reader handle will now have to `Arc::clone(&self.db)` and pass it to the actual `rayon` thread as they don't hold any state themselves.

This means we're adding an `Arc::clone` each request.

Reply:
This should be ok